scripts/remote_eval.py

```diff
@@ -0,0 +1,192 @@
+import dataclasses
+import datetime
+import faulthandler
+import os
+
+import numpy as np
+import pandas as pd
+import tqdm
+import tyro
+from moviepy.editor import ImageSequenceClip
+from PIL import Image
+
+from droid.robot_env import RobotEnv
+from scripts import websocket_policy_client
+
+faulthandler.enable()
+
+
+@dataclasses.dataclass
+class Args:
+    # Required arguments
+    policy_name: str
+
+    # Hardware parameters
+    left_camera_id: str = "24259877"
+    right_camera_id: str = "24514023"
+    wrist_camera_id: str = "13062452"
+
+    # Rollout parameters
+    max_timesteps: int = 400
+    # How many actions to execute from a predicted action chunk before querying policy server again
+    open_loop_horizon: int = 8
+
+    # Remote server parameters
+    remote_host: str = "localhost"
+    remote_port: int = 8000
+
+
+def main(args: Args):
+    # Initialize the Panda environment.
+    env = RobotEnv(action_space="joint_velocity", gripper_action_space="position")
+    print("created the droid env!")
+
+    policy_client = websocket_policy_client.WebsocketClientPolicy(args.remote_host, args.remote_port)
+
+    df = pd.DataFrame(columns=["policy_name", "success", "duration", "video_filename"])
+
+    while True:
+        # Rollout parameters
+        actions_from_chunk_completed = 0
+        pred_action_chunk = None
+
+        # Prepare to save video of rollout
+        timestamp = datetime.datetime.now().strftime("%Y_%m_%d_%H:%M:%S")
+        video = []
+        bar = tqdm.tqdm(range(args.max_timesteps))
+        for t_step in bar:
+            try:
+                # Get the current observation
+                curr_obs = _extract_observation(
+                    args,
+                    env.get_observation(),
+                    # Save the first observation to disk
+                    save_to_disk=t_step == 0,
+                )
+
+                video.append(curr_obs["right_image"])
+
+                # Send websocket request to policy server if it's time to predict a new chunk
+                if actions_from_chunk_completed == 0 or actions_from_chunk_completed >= args.open_loop_horizon:
+                    actions_from_chunk_completed = 0
+
+                    request_data = {
```

Comment:
I think the script currently lacks the functionality to get a language instruction from the user

Reply:
Yeah we have the language instruction on the server side currently because it gets baked into the tokenizer. Changing this will be a bit tricky so I think we're putting it off for now.